src/pydap/model.py

```diff
@@ -526,18 +530,18 @@ def __getitem__(self, key):
 
         # If it's a tuple, return a new `SequenceType` with selected children.
         elif isinstance(key, tuple):
-            out = SequenceType(self.name, self.data, self.attributes.copy())
+            out = self.__class__(self.name, self._data, self.attributes.copy())
```

Comment:
nit: usually I write `type(self)` instead of `self.__class__`, but I think they do the same thing.